dso-l2/src/main/java/com/tc/l2/state/StateManagerImpl.java

```diff
@@ -210,7 +210,12 @@ private synchronized void moveToPassiveReady(Enrollment winningEnrollment) {
     logger.debug("moving to passive ready " + state + " " + winningEnrollment);
     if (state == START_STATE) {
       setActiveNodeID(winningEnrollment.getNodeID());
-      state = PASSIVE_UNINITIALIZED;
+      state = (startState == null) ? PASSIVE_UNINITIALIZED : startState;
+      if (!state.equals(PASSIVE_STANDBY) && !state.equals(PASSIVE_UNINITIALIZED)) {
+// TODO:  make sure this is the proper way to handle this.
```

Comment:
Do we even hit this code path in case of restartable server?
````
} else if (activeNode.isNull() || activeNode.equals(winningEnrollment.getNodeID())
               || clusterMsg.getType() == L2StateMessage.ELECTION_WON) {
      // There is no active server for this node or the other node just detected a failure of ACTIVE server and ran an
      // election and is sending the results. This can happen if this node for some reason is not able to detect that
      // the active is down but the other node did. Go with the new active.
      if (startState == null || startState == START_STATE) {
        moveToPassiveReady(winningEnrollment);
        if (clusterMsg.getType() == L2StateMessage.ELECTION_WON_ALREADY) {
          sendOKResponse(clusterMsg.messageFrom(), clusterMsg);
        }
      } else {
        //this server was started with persistent data, this server could be either a ACTIVE or PASSIVE before
        // it went down. In any case, don't start this server in passive state.
// TODO: send a negative response so the active zaps this server.  This is a bad way to agree.  find a better way
        if (clusterMsg.getType() == L2StateMessage.ELECTION_WON_ALREADY) {
          sendNGResponse(clusterMsg.messageFrom(), clusterMsg);
        }
      }
    } else {
````